imports/ui/components/partyAdd/client/partyAdd.tests.js

```diff
@@ -21,75 +25,114 @@ import {
-        stubs.restoreAll();
+
+        if (!user.create) {
+            Accounts.createUser({
+                username: user.username,
+                password: user.password
+            }, function(error) {
+                user.create = true;
+                if (!Meteor.userId()) {
+                    Meteor.loginWithPassword(user.username, user.password, function() {
+                        user._id = Meteor.userId();
+                        done();
+                    });
+                } else {
+                    user._id = Meteor.userId();
+                    done();
+                }
+            });
+        } else {
+            done();
+        }
     });
 
     // Test inside controller
     describe('controller', function() {
         let controller;
+
         const party = {
             name: 'Foo',
             description: 'Birthday of Foo',
             public: true
         };
-        const user = {
-            _id: 'userId'
-        }
+        const doneCallback = function() {};
 
         // Initialize controller
-        beforeEach(function() {
+        beforeEach(function(done) {
             inject(function($rootScope, $componentController) {
                 controller = $componentController(PartyAdd, {
                     $scope: $rootScope.$new(true)
+                }, {
+                    done: doneCallback
                 });
             });
-
-            if (stubs.user)
-                stubs.user.restore();
-
-            stubs.create('user', Meteor, 'user').returns(user);
+            done();
         });
 
         describe('reset()', function() {
-            it('should clean up party object', function() {
+            it('should clean up party object', function(done) {
                 controller.party = party;
                 controller.reset();
 
                 expect(controller.party).to.be.deep.equal({});
+                done();
             });
         });
 
         describe('submit()', function() {
             // Monitors insert, reset on submit calls
-            beforeEach(function() {
-                if (spies.insert)
-                    spies.insert.restore();
-                if (spies.reset)
-                    spies.reset.restore();
-
+            beforeEach(function(done) {
                 spies.create('insert', Parties, 'insert');
                 spies.create('reset', controller, 'reset');
 
                 controller.party = party;
 
-                controller.submit();
+                if (!Meteor.userId()) {
+                    Meteor.loginWithPassword(user.username, user.password, function() {
+                        controller.submit();
+                        done();
+                    });
+                } else {
+                    controller.submit();
+                    done();
+                }
+            });
+
+            afterEach(function(done) {
+                if (spies.insert)
+                    spies.insert.restore();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'spies'.](https://www.codacy.com/app/ddspog/MAM-Seed/file/2729684806/issues/source?bid=3340041&fileBranchId=3346427#l116)